src/main/java/monique/parser/Parser.java

```diff
@@ -20,136 +20,135 @@
-                        Task taskToAdd = new Deadline(description, false, by);
-                        command = new AddCommand(taskToAdd);
-                        break;
-                    } catch (ParseException pe) {
-                        pe.advice();
-                    } catch (IllegalDateFormatException idee) {
-                        idee.advice();
+            }
+            case "deadline": {
+                try {
+                    String[] parts = fullCommand.split("/by");
+                    if (parts.length <= 1) {
+                        throw new ParseException();
                     }
+                    String by = parts[1].trim();
+                    String[] commandAndDescription = parts[0].trim().split(" ", 2);
+                    String description = commandAndDescription.length > 1 ? commandAndDescription[1] : "";
+                    Task taskToAdd = new Deadline(description, false, by);
+                    command = new AddCommand(taskToAdd);
+                    break;
+                } catch (ParseException pe) {
+                    pe.advice();
+                } catch (IllegalDateFormatException idee) {
+                    idee.advice();
                 }
-                case "event": {
-                    try {
-                        String[] fromSplit = fullCommand.split("/from");
-                        if (fromSplit.length!= 2){
-                            throw new ParseException();
-                        }
-                        String[] toSplit = fromSplit[1].split("/to");
-                        if (toSplit.length != 2) {
-                            throw new ParseException();
-                        }
-                        String[] commandAndDescription = fromSplit[0].trim().split(" ", 2);
-                        String description = commandAndDescription.length > 1 ? commandAndDescription[1] : "";
-                        String fromDate = toSplit[0].trim();
-                        String toDate = toSplit[1].trim();
-                        Task taskToAdd = new Event(description, false, fromDate, toDate);
-                        command = new AddCommand(taskToAdd);
-                        break;
-                    } catch (ParseException pe) {
-                        pe.advice();
-                    } catch (IllegalDateFormatException idee) {
-                        idee.advice();
+            }
+            case "event": {
+                try {
+                    String[] fromSplit = fullCommand.split("/from");
+                    if (fromSplit.length != 2) {
+                        throw new ParseException();
+                    }
+                    String[] toSplit = fromSplit[1].split("/to");
+                    if (toSplit.length != 2) {
+                        throw new ParseException();
                     }
+                    String[] commandAndDescription = fromSplit[0].trim().split(" ", 2);
```

Comment:
Excellent work handling the parsing of complex user inputs, especially for event commands with /from and /to delimiters. The use of clear exception handling to manage parsing issues is a strong approach that will help prevent user errors from crashing the program.